src/main/java/frc/robot/subsystems/swerve/states/heading/HeadingStabilizer.java

```diff
@@ -0,0 +1,41 @@
+package frc.robot.subsystems.swerve.states.heading;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Rotation2d;
+import frc.constants.MathConstants;
+import frc.robot.subsystems.swerve.SwerveConstants;
+
+public class HeadingStabilizer {
+
+	private final PIDController headingController;
+	private boolean targetLocked;
```

Comment:
you need to initialize this so there won't be a way to cause an error if you call it before you set it

Reply:
initialize the controller? its in the constructor

Reply:
The targetLocked

Reply:
its false by default by i will init in the constructor